app/api/utility.py

```diff
@@ -217,24 +232,21 @@ def create_query(
             {{
                 SELECT ?subject (count(distinct ?phenotypic_session) as ?num_matching_phenotypic_sessions)
                 WHERE {{
-                    ?subject a nb:Subject.
-                    OPTIONAL {{
-                        ?subject nb:hasSession ?phenotypic_session.
-                        ?phenotypic_session a nb:PhenotypicSession.
-
-                        OPTIONAL {{?phenotypic_session nb:hasAge ?age.}}
-                        OPTIONAL {{?phenotypic_session nb:hasSex ?sex.}}
-                        OPTIONAL {{?phenotypic_session nb:hasDiagnosis ?diagnosis.}}
-                        OPTIONAL {{?phenotypic_session nb:isSubjectGroup ?subject_group.}}
-                        OPTIONAL {{?phenotypic_session nb:hasAssessment ?assessment.}}
-                    }}
+                    ?subject nb:hasSession ?phenotypic_session.
+                    ?phenotypic_session a nb:PhenotypicSession.
+
+                    OPTIONAL {{?phenotypic_session nb:hasAge ?age.}}
+                    OPTIONAL {{?phenotypic_session nb:hasSex ?sex.}}
+                    OPTIONAL {{?phenotypic_session nb:hasDiagnosis ?diagnosis.}}
+                    OPTIONAL {{?phenotypic_session nb:isSubjectGroup ?subject_group.}}
+                    OPTIONAL {{?phenotypic_session nb:hasAssessment ?assessment.}}
+
                     {phenotypic_session_level_filters}
                 }} GROUP BY ?subject
             }}
             {{
                 SELECT ?subject (count(distinct ?imaging_session) as ?num_matching_imaging_sessions)
                 WHERE {{
-                    ?subject a nb:Subject.
```

Comment:
Are we removing this cause it's unnecessary to do since the only nodes that have a `nb:hasSession` edge to nodes of class `nb:ImagingSession` are of class `nb:Subject`?

Reply:
This statement can be removed because the main query (which comes first) already filters for subjects as being of type `nb:Subject`

https://github.com/neurobagel/api/blob/0ac0d4297d0942e555dadd066555540341cf6156/app/api/utility.py#L196-L202

so in subsequent subqueries we should be able to simply reuse the resulting `?subject` without having to repeat the type check again! This just makes the query a bit more efficient.